integration-tests/test/fee-payment.spec.ts

```diff
@@ -4,37 +4,183 @@ chai.use(chaiAsPromised)
 
 /* Imports: External */
 import { ethers, BigNumber, Contract, utils } from 'ethers'
-import { TxGasLimit, TxGasPrice } from '@eth-optimism/core-utils'
-import { predeploys, getContractInterface } from '@eth-optimism/contracts'
+import { sleep } from '@eth-optimism/core-utils'
+import { serialize } from '@ethersproject/transactions'
+import {
+  predeploys,
+  getContractInterface,
+  getContractFactory,
+} from '@eth-optimism/contracts'
 
 /* Imports: Internal */
-import { IS_LIVE_NETWORK } from './shared/utils'
+import { IS_LIVE_NETWORK, expectApprox } from './shared/utils'
 import { OptimismEnv } from './shared/env'
 import { Direction } from './shared/watcher-utils'
 
+const setPrices = async (env: OptimismEnv, value: number | BigNumber) => {
+  const gasPrice = await env.gasPriceOracle.setGasPrice(value)
+  await gasPrice.wait()
+  const baseFee = await env.gasPriceOracle.setL1BaseFee(value)
+  await baseFee.wait()
+}
+
 describe('Fee Payment Integration Tests', async () => {
   let env: OptimismEnv
+  const other = '0x1234123412341234123412341234123412341234'
+
   before(async () => {
     env = await OptimismEnv.new()
   })
 
-  let ovmSequencerFeeVault: Contract
-  before(async () => {
-    ovmSequencerFeeVault = new Contract(
-      predeploys.OVM_SequencerFeeVault,
-      getContractInterface('OVM_SequencerFeeVault'),
-      env.l2Wallet
+  it(`should return eth_gasPrice equal to OVM_GasPriceOracle.gasPrice`, async () => {
+    const assertGasPrice = async () => {
+      const gasPrice = await env.l2Wallet.getGasPrice()
+      const oracleGasPrice = await env.gasPriceOracle.gasPrice()
+      expect(gasPrice).to.deep.equal(oracleGasPrice)
+    }
+
+    assertGasPrice()
+    // update the gas price
+    const tx = await env.gasPriceOracle.setGasPrice(1000)
+    await tx.wait()
+
+    assertGasPrice()
+  })
+
+  it('Paying a nonzero but acceptable gasPrice fee', async () => {
+    await setPrices(env, 1000)
+
+    const amount = utils.parseEther('0.0000001')
+    const balanceBefore = await env.l2Wallet.getBalance()
+    const feeVaultBalanceBefore = await env.l2Wallet.provider.getBalance(
+      env.sequencerFeeVault.address
+    )
+    expect(balanceBefore.gt(amount))
+
+    const tx = await env.ovmEth.transfer(other, amount)
+    const receipt = await tx.wait()
+    expect(receipt.status).to.eq(1)
+
+    const balanceAfter = await env.l2Wallet.getBalance()
+    const feeVaultBalanceAfter = await env.l2Wallet.provider.getBalance(
+      env.sequencerFeeVault.address
+    )
+
+    const l2Fee = receipt.gasUsed.mul(tx.gasPrice)
+
+    const raw = serialize({
+      nonce: tx.nonce,
+      value: tx.value,
+      gasPrice: tx.gasPrice,
+      gasLimit: tx.gasLimit,
+      to: tx.to,
+      data: tx.data,
+    })
+
```

Comment:
This needs a comment that R S and V are not rlp encoded as 0. The RLP list is 6 elements